libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -32,11 +32,81 @@
 #import "SFApplicationHelper.h"
 #import "SFSecurityLockout.h"
 
+@interface SFSDKUIWindow ()
+- (instancetype)initWithFrame:(CGRect)frame;
+- (instancetype)initWithFrame:(CGRect)frame andName:(NSString *)windowName;
+- (void)stashRootViewController;
+- (void)unstashRootViewController;
+
+@property (nonatomic,strong) UIViewController * _Nullable stashedController;
+@property (nonatomic,readonly) NSString * _Nullable windowName;
+@end
+
+@implementation SFSDKUIWindow
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        _windowName = @"NONAME";
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame andName:(NSString *)windowName {
+    self = [super initWithFrame:frame];
+    if (self) {
+        _windowName = windowName;
+    }
+    return self;
+}
+
+- (void)stashRootViewController {
+    if (self.rootViewController) {
+        _stashedController = self.rootViewController;
+        super.rootViewController = nil;
+    }
+}
+
+- (void)unstashRootViewController {
```

Comment:
This unstash function is called when the window is made key window. UIKIT invokes the UIWindow:becomeKeyWindow. This is when we unstash (set) the rootviewcontroller.